components/omega/doc/design/OmegaV0ShallowWater.md

```diff
@@ -0,0 +1,660 @@
+$F_e = u_e [h_i \phi_i]_e$
+ in the tracer advection term. To be clear, when we refer to the divergence variable, rather than the operator, this means
+$$
+\left( \nabla \cdot {\bf u}\right)_i =
+D_i
+=  \frac{1}{A_i} \sum_{e\in EC(i)} n_{e,i} \, u_e \, l_e.
+$$
+
+
+#### 3.2.2. Gradient
+
+The gradient operator maps a cell-centered scalar to an edge-normal vector component
+([Ringler et al. 2010](https://www.sciencedirect.com/science/article/pii/S0021999109006780) eqn 22) is
+
+$$
+\left( \nabla h \right)_e
+\equiv
+\nabla h_i = \frac{1}{d_e} \sum_{i\in CE(e)} -n_{e,i} \, h_i,
+$$
+where $i\in CE(e)$ identifies the two cells that neighbor an edge. The edge subscript is dropped, so $\nabla h_i$ is understood to occur at an edge.
+
+For practical purposes the sign indicator variable $n_{e,i}$ is not used in the code, and we opt for the simpler formulation
+$$
+\left( \nabla h \right)_e
+\equiv
+\nabla h_i = \frac{h_{i2} - h_{i1} }{d_e},
+$$
+where $\left\{i_1, i_2\right\}$ are the cells neighboring edge $e$.  The indices $\left\{i_1, i_2\right\}$ are ordered such that the normal vector ${\bf n}$ points from cell $i_1$ to cell $i_2$. In the code ${\bf n}$ points from CellsOnEdge(0,iEdge) to CellsOnEdge(1,iEdge).
+
+#### 3.2.3. Curl
+The curl operator maps a vector's edge normal component $u_e$ to a scalar field at vertex
+([Ringler et al. 2010](https://www.sciencedirect.com/science/article/pii/S0021999109006780) eqn 23) and is
+
+$$
+\left( {\bf k} \cdot \left( \nabla \times {\bf F} \right)\right)_v
+\equiv
+{\bf k} \cdot \left( \nabla \times F_e \right)
+= \frac{1}{A_v} \sum_{e\in EV(v)} t_{e,v} \, F_e \, d_e
+$$
+
+where $A_v$ is the area of the dual mesh cell $v$, i.e. the triangle surrounding vertex $v$. Similar to $n_{e,i}$, the indicator function $t_{e,v}$ tracks whether a positive $F_e$ makes a positive or negative contribution to the curl function at vertex $v$. If the vector ${\bf k} \times {\bf n}_e$ is directed toward
+${\bf x}_v$, then $t_{e,v}=1$. The summation is over $e\in EV(v)$, which are the edges that terminate at vertex $v$. There are *always* three edges that terminate at each vertex for Voronoi Tessellations, and four edges on each vertex for quadrilateral meshes, unless neighboring cells are missing for land boundaries. Again, the subscript $v$ is dropped and a vertex is assumed for the location of the curl.
+
+
+#### 3.2.4. Perpendicular vector component
+
+The perpendicular vector component a vector field is defined as
+$$
+ \nabla^\perp {\bf F} \equiv
+ {\bf k} \times \nabla {\bf F}
+$$
+so that it points 90$^o$ to the left of ${\bf F}$ in a right-handed coordinate system. 
+
+In TRiSK, the native component for the prognostic velocity field is normal edge velocity $u_e$. The perpendicular component at the edge is computed diagnostically as
+([Ringler et al. 2010](https://www.sciencedirect.com/science/article/pii/S0021999109006780) eqn 24)
+
+$$
+u_e^\perp = \frac{1}{d_e} \sum_{e'\in ECP(e)} w_{e,e'} \, l_e \, u_{e'}.
```

Comment:
The index on `l` should be `e'`. There is also another issue. While this agrees with the TRiSK paper, it does not agree with Omega code. I believe that in MPAS-Ocean mesh files the reconstruction weights are pre-multiplied by `l_e' / d_e`. It would be good to make a note about this here. The weights also need to be added to the variables table.

Reply:
Thanks for your detailed review! I fixed this and added text to explain what is used in the code. See the [revised section in the compiled version here](https://portal.nersc.gov/project/e3sm/mpeterse/omega/design-omega0-sw/html/design/OmegaV0ShallowWater.html#perpendicular-vector-component).